### DISCUSSION TITLE
Model Card Metadata generation from GGUF KV store

### DISCUSSION THREAD

**Discussion by mofosyne** [ggml-org/llama.cpp#10304] [Ideas]
Right now we can deduce to a certain extent from HF model cards the metadata that should go into a GGUF file.

But maybe we should have a mode to read a GGUF file and generate a model card template? Wonder if that's needed or not.

**Comment by Thomas-MMJ** [comment]


**Reply by mofosyne** [reply]
Nice tip.

Well if anyone decides to take that thread further along, they may want to check with this page

https://github.com/ggerganov/llama.cpp/wiki/HuggingFace-Model-Card-Metadata-Interoperability-Consideration

As that will contain the agreed upon mapping between llama.cpp and huggingface about how the model card metadata will map to gguf kv store and vis versa.

**Reply by Thomas-MMJ** [reply]
Deleted my comment since the function dump_metadata_json is better,

https://github.com/ggerganov/llama.cpp/blob/master/gguf-py/gguf/scripts/gguf_dump.py
